### PR TITLE
Add BC shim for template lookup in former directories.

### PR DESCRIPTION
**Pull request by dereuromark**
Follow https://github.com/propelorm/Propel2/pull/1848

Just to make sure we ease the upgrade issues here for folks that use behaviors or other custom repositories with extension and templates.

### CODE REVIEW

**Comment by dereuromark** [comment]
Could someone dev-qa and approve this please?
@mringler @oojacoboo etc

**Comment by oojacoboo** [comment]
Logic seems okay, but I don't fully know what input and output to expect here, so it's hard to give it much scrutiny.

File: src/Propel/Common/Util/PathTrait.php [hunk 1]
```diff
@@ -36,7 +36,15 @@ protected function getTemplatePath(string $path): string
     {
         $srcPos = strpos($path, DIRECTORY_SEPARATOR . 'src' . DIRECTORY_SEPARATOR);
         if ($srcPos === false) {
```
Comment (mringler) [inline src/Propel/Common/Util/PathTrait.php:38]: I am probably misunderstanding this, but above, it says templates can be on the same level as src folder, i.e. `./Propel2/templates`, which does not contain `/src/`. Wouldn't that throw an error here, even though it is a valid path?

**Review by mringler — APPROVED** [review]

**Comment by dereuromark** [comment]
You can see the linked pr for previous template paths

**Review by mringler — APPROVED** [review]
Great, really good to give users feedback why something isn't working.

In general, the extraction of the path elements (`$elements`) could be done a bit more readable (I think) as
```php
        $lastSrcPos = strrpos($path, DIRECTORY_SEPARATOR . 'src' . DIRECTORY_SEPARATOR);
        $pathAfterSrc = substr($path, $lastSrcPos + 5);
        $elements = explode(DIRECTORY_SEPARATOR, $pathAfterSrc);
```
but that's fine

**Pull request merged**